dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry/internal/cdi/WithSpanInterceptor.java

```diff
@@ -75,52 +86,63 @@ private static SpanKind spanKindFromMethod(Method method) {
+                Tr.debug(this, tc, "Method " + invocationContext.getMethod().toString() + " Should start: " + shouldStart);
             }
-        }
 
-        try {
-            Object result = invocationContext.proceed();
             if (shouldStart) {
-                instrumenter.end(spanContext, methodRequest, null, null);
+                spanContext = instrumenter.start(parentContext, methodRequest);
+                scope = spanContext.makeCurrent();
                 if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
-                    Tr.debug(this, tc, "spanContext " + spanContext.toString() + " has ended");
+                    Tr.debug(this, tc, "spanContext " + spanContext.toString() + " has started and is now the current context");
                 }
             }
-            return result;
-        } finally {
-            if (scope != null) {
-                scope.close();
+
+            try {
+                Object result = invocationContext.proceed();
+                if (shouldStart) {
+                    instrumenter.end(spanContext, methodRequest, null, null);
+                    if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                        Tr.debug(this, tc, "spanContext " + spanContext.toString() + " has ended");
+                    }
+                }
+                return result;
+            } finally {
+                if (scope != null) {
+                    scope.close();
+                }
             }
-        }
+        } catch (Exception e) {
+            Tr.error(tc, Tr.formatMessage(tc, "CEMOT5101.telemetry.error", e));
+            return null;
+        } 
     }
 
     private static final class MethodRequestSpanNameExtractor implements SpanNameExtractor<MethodRequest> {
         @Override
         public String extract(final MethodRequest methodRequest) {
-            return AccessController.doPrivileged((PrivilegedAction<String>) () -> {
-                WithSpan annotation = methodRequest.getMethod().getDeclaredAnnotation(WithSpan.class);
-                String spanName = annotation.value();
-                if (spanName.isEmpty()) {
-                    spanName = SpanNames.fromMethod(methodRequest.getMethod());
-                }
-                return spanName;
-            });
+            try {
+                return AccessController.doPrivileged((PrivilegedAction<String>) () -> {
+                    WithSpan annotation = methodRequest.getMethod().getDeclaredAnnotation(WithSpan.class);
+                    String spanName = annotation.value();
+                    if (spanName.isEmpty()) {
+                        spanName = SpanNames.fromMethod(methodRequest.getMethod());
+                    }
+                    return spanName;
+                });
+            } catch (Exception e) {
+                Tr.error(tc, Tr.formatMessage(tc, "CEMOT5101.telemetry.error", e));
+                return null;
+            } 
```

Comment:
Don't need to do this method since it's called within the try-catch added for `span()`

Reply:
This change still needs done.